.github/scripts/update-sources.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+source `dirname ${BASH_SOURCE[0]}`/config.sh
+
+function update_repository() {
+    DIRECTORY=$1
+    REPOSITORY=$2
+    BRANCH=$3
+    if ! [ -d $DIRECTORY ]; then
+        git clone $REPOSITORY -b $BRANCH --single-branch --depth 1 $DIRECTORY
+        git submodule init
+        git submodule update
+    else
+        pushd $DIRECTORY
```

Comment:
I would prefer to have a switch in build.sh and in this script which will allow to reset the repo. Otherwise, just skip.

Reply:
It's a good idea to add the possibility to disable the reset. It should not be a default though because then the update will frequently fail, and it won't produce reproducible build.

Reply:
The `RESET_SOURCES` option and comments for all the `config.sh` options added.